### DISCUSSION TITLE
How do I use SAMPA with Coqui?

### DISCUSSION THREAD

**Discussion by NoteToSelfFindGoodNickname** [coqui-ai/TTS#2112] [General Q&A]
Hello!

There is a discussion [here](https://github.com/coqui-ai/TTS/discussions/503) about how to IPA with Coqui.

However, I would like to use an extended version of SAMPA:

k I1 n - d r= - g a2 6 - t n=
(meaning "kindergarten")

    "k" = "k"
    "I" = "i", 1 means primarily accented
    "n" = "n" 
    "-" = syllable break
    "d" = "d"
    "r=" = "er"
    "-" = syllable break
    "g" = "g"
    "a2" = "a", 2 means secondarily accented
    "6" = "r"
    "-" = syllable break
    "t" = "t"
    "n=" = "en"

I guess I could convert them to IPA symbols, but how should I represent the syllable breaks ("-") and the accents?

Or should I treat the "E", "E1" and "E2" as different characters and even "-" as a new character?

Instead of using IPA, I could just use some arbitrary unicode characters, right?
For example:

```
- = chrw(4000)
a = chrw(4001)
a1 = chrw(4002)
a2 = chrw(4003)
b = chrw(4004)
c = chrw(4005)
C = chrw(4006)
d = chrw(4007)
D = chrw(4008)
e = chrw(4009)
e1 = chrw(4010)
e2 = chrw(4011)
E1 = chrw(4012)
etc.
```

Right?

Thanks for any input.

**Comment by NoteToSelfFindGoodNickname** [comment]
Bump

**Comment by erogol** [comment]
All text processing is done in TTS.tts.utils.tokenizer 

So I'd suggest checking it and overriding components as needed for SAMPA
